### DISCUSSION TITLE
Print not working, having to use sys.stdout.flush()

### DISCUSSION THREAD

**Discussion by abel-diaz** [sfu-db/connector-x#135] [Q&A]
HI,
The ability to use print() is not working. I have to add "sys.stdout.flush()" to see the print statement when using connector-x. 
Ex:
print('starting connector x')
sys.stdout.flush()

code....

print('connector x done')
sys.stdout.flush()

It would be nice to not have to use the additional code of sys.stdout.flush() 

Thank you!
Abel

**Comment by wangxiaoying** [comment]
Hi @abel-diaz , I cannot reproduce your problem on my local machine. Can you list your environment setup? Also when will this issue happen? e.g. Whenever connectorx is imported or only after `cx.read_sql` is invoked?

**Reply by abel-diaz** [reply]
Hi, it seems that it happens only in jupyter notebook in the cell block that cx is used. It works fine but if ran again, it wont output. I dont have this problem in Atom or 1st runs in JPNB. Seems to be only in re-runs in JPNB.

**Reply by wangxiaoying** [reply]
That's weird, I will look into it. Thanks for the report!

In the meanwhile, you can use `print('starting connector x', flush=True)` instead of `sys.stdout.flush()` to simplify the code for now.
